file/file.go

```diff
@@ -397,3 +397,90 @@ func HasPermission(fileName string, permission int) (bool, error) {
 
 	return true, nil
 }
+
+func CountLine(filename string) (uint64, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var lineCount uint64
+	for scanner.Scan() {
+		lineCount++
+	}
+
+	if err := scanner.Err(); err != nil {
+		return 0, err
+	}
+	return lineCount, nil
+}
+
+func CountLineWithSeparator(separator, filename string) (uint64, error) {
+	if separator == "" || len(separator) > 1 {
+		return 0, errors.New("invalid separator")
+	}
+	file, err := os.Open(filename)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	searchBytes := []byte(separator)
+	searchLen := len(searchBytes)
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		dataLen := len(data)
+		// Return nothing if at end of file and no data passed
+		if atEOF && dataLen == 0 {
+			return 0, nil, nil
+		}
+		// Find next separator and return token
+		if i := bytes.Index(data, searchBytes); i >= 0 {
+			return i + searchLen, data[0:i], nil
+		}
+		// If we're at EOF, we have a final, non-terminated line. Return it.
+		if atEOF {
+			return dataLen, data, nil
+		}
+		// Request more data.
+		return 0, nil, nil
+	})
+
+	var count uint64
+	for scanner.Scan() {
+		count++
+	}
+	return count, scanner.Err()
+}
+
+type FileInfo struct {
+	Filename  string
+	LineCount uint64
+}
+
+func CountLinesInMultipleFiles(filenames []string) ([]FileInfo, error) {
+	var fileInfos []FileInfo
+	for _, filename := range filenames {
+		lineCount, err := CountLine(filename)
+		if err != nil {
+			return nil, err
+		}
+		fileInfos = append(fileInfos, FileInfo{Filename: filename, LineCount: lineCount})
+	}
+	return fileInfos, nil
+}
+
+func CountTotalLinesInMultipleFiles(filenames []string) (uint64, error) {
```

Comment:
This function seems redundant, I think it can be removed as users can aggregate data with previous funcs